polaris-service/src/test/java/org/apache/polaris/service/test/TestEnvironment.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.test;
+
+import jakarta.ws.rs.client.Client;
+import java.net.URI;
+
+/** Defines the test environment that a test should run in. */
+public record TestEnvironment(Client apiClient, String baseUrl) {
+  public URI getBaseUri() {
```

Comment:
You could make the main constructor take a URI and have a secondary constructor that takes the string and converts to a URI. That way the conversion is done on construction and always done once.

Reply:
Updated